src/main/java/at/hannibal2/skyhanni/features/itemabilities/CrownOfAvariceCounter.kt

```diff
@@ -39,28 +45,44 @@ object CrownOfAvariceCounter {
     private var display: List<Renderable> = emptyList()
     private val MAX_AVARICE_COINS = 1.billion
     private val MAX_AFK_TIME = 2.minutes
+    private var inventoryOpen = false
     private val isWearingCrown by RecalculatingValue(1.seconds) {
         InventoryUtils.getHelmet()?.getInternalNameOrNull() == internalName
     }
 
     private var count: Long? = null
     private var coinsEarned: Long = 0L
-    private var sessionStart: SimpleTimeMark? = null
+    private var sessionStart: Duration? = null
     private var lastCoinUpdate: SimpleTimeMark? = null
-    private val isSessionActive get(): Boolean = sessionStart?.passedSince()?.let { it < config.sessionActiveTime.seconds } ?: false
+    private var isFrozen: Boolean = false
+    private val isSessionActive get(): Boolean = sessionStart?.let { it < config.sessionActiveTime.seconds } ?: false
     private var coinsDifference: Long? = null
 
-    @HandleEvent
-    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
-        if (!isEnabled()) return
-        if (!isWearingCrown) return
-        config.position.renderRenderables(display, posLabel = "Crown of Avarice Counter")
+    init {
+        RenderDisplayHelper(
+            outsideInventory = true,
+            inOwnInventory = true,
+            condition = { isEnabled() && isWearingCrown },
+            onRender = { renderDisplay(config.position) }
+        )
     }
 
+    fun renderDisplay(pos: Position) {
+        val invCurrentlyOpen = Minecraft.getMinecraft().currentScreen?.let { it is GuiInventory || it is GuiChest } ?: false
+        if (inventoryOpen != invCurrentlyOpen) {
+            inventoryOpen = invCurrentlyOpen
+            update()
+        }
+
+        pos.renderRenderables(display, posLabel = "Crown of Avarice Counter")
+    }
+
+
     @HandleEvent
     fun onSecondPassed(event: SecondPassedEvent) {
         if (!isEnabled()) return
         if (!isWearingCrown) return
+        if (!isSessionAFK()) sessionStart = sessionStart?.plus(1.seconds)
```

Comment:
If sessionStart isn't nullable this can be
```suggestion
        if (!isSessionAFK()) sessionStart += 1.seconds
```